validation/1.4/modules/ROOT/pages/validation-examples.adoc

```diff
@@ -0,0 +1,583 @@
+* *White list* +
+`Edit inline`
+* *Ips* +
+`Edit inline`
++
+[start=4]
+. Click the plus sign (*+*) to open the IP value window.
+. Set the *Value* field to `192.168.1.0/24`.
+. Repeat the previous two steps and add the following IP addresses `127.0.0.1` and `193.1`.
+
+.Is whitelisted ip Configuration
+image::validation-iswhitelistedip.png[Is whitelisted ip Configuration]
+
+In the XML editor, the configuration looks like this:
+
+[source,xml,linenums]
+----
+<flow name="FlowValidation" >
+		<validation:is-whitelisted-ip ipAddress="payload">
+			<validation:white-list >
+				<validation:ips >
+					<validation:ip value="192.168.1.0/24" />
+					<validation:ip value="127.0.0.1" />
+					<validation:ip value="193.1" />
+				</validation:ips>
+			</validation:white-list>
+		</validation:is-whitelisted-ip>
+	</flow>
+----
+
+== Configure the Matches Regex Operation
+
+The *Matches regex* operation validates that a given expression matches a Java regular expression.
+
+The following example shows how to configure the operation in Anypoint Studio:
+
+. In the *Mule Palette* view, select *Validation > Matches regex*.
+. Drag *Matches regex* to the Studio canvas.
+. In the *Matches regex* configuration screen, for the *Expression* field select `Expression`, and in the empty field add  `#[vars.doReply]`.
+
+.Matches regex Configuration
+image::validation-matchesregex.png[Matches regex Configuration]
+
+In the XML editor, the configuration looks like this:
+
+[source,xml,linenums]
+----
+<validation:matches-regex value="#[payload]" regex=".+\.log" caseSensitive="false"/>
+----
+
+== Configure the Validate Size Operation
+
+The *Validate size* operation validates that the input size is between the given minimum and maximum boundaries.
+This is valid for inputs of type String, Collection, Map, and Array. In the case of a String, the size refers to the length in characters.
+
+The following example shows how to configure the operation in Anypoint Studio:
+
+. In the *Mule Palette* view, select *Validation > Validate size*.
+. Drag *Validate size* to the Studio canvas.
+. In the *Validate size* configuration screen, complete the following parameters:
+
+* *Value* +
+`payload`
+* *Min* +
+`vars.minLength`
+* *Max* +
+`vars.maxLength`
+
+.Validate size Configuration
+image::validation-validatesize.png[Validate size Configuration]
+
+In the XML editor, the configuration looks like this:
+
+[source,xml,linenums]
+----
+<validation:validate-size value="#[payload]" min="#[vars.minLength]" max="#[vars.maxLength]"/>
+----
+
+== Configure the All Scope
+
+The *All* scope evaluates a list of nested validation operations and informs only one `VALIDATION:MULTIPLE` error which summarizes all of the found errors.
+There are scenarios in which you may want to evaluate several conditions, out of which more than one could fail simultaneously. In these cases, it’s ideal to generate a single error that contains all of the descriptions. About the *All* scope validator:
```

Comment:
```suggestion
Sometimes you want to evaluate several conditions, and some might fail simultaneously. In these cases, generate a single error that contains all of the descriptions. About the *All* scope validator:
```

Reply:
Hmm...so it seems that this para clarifies the one previous, that Diane commented about. 

I think we are repeating the same thing in three ways above, here, and below.

Can we...:

- Explain how this is a SCOPE and not an OPERATION, like the previous subtopics
- Say that our audience should configure the All scope to validate several conditions but receive just one multiline error message if any or all conditions fail (one line per error)?
- Why do we have to say that validations are executed (?) even if all fail: don't you have to validate before you can even tell if something fails? What is executed?